server/tests/test_log_sanitizer.py

```
@@ -0,0 +1,190 @@
+    def test_strips_carriage_return(self):
+        assert sanitize("line1\rline2") == "line1line2"
+
+    def test_strips_tab(self):
+        assert sanitize("col1\tcol2") == "col1col2"
+
+    def test_strips_null_byte(self):
+        assert sanitize("before\x00after") == "beforeafter"
+
+    def test_strips_unicode_line_separator(self):
+        """U+2028 LINE SEPARATOR can forge new log lines in Unicode parsers."""
+        assert sanitize("before after") == "beforeafter"
```

Ruff (0.15.11)

[warning] String contains ambiguous ` ` (LINE SEPARATOR). Did you mean ` ` (SPACE)?

(RUF001)